src/index.js

```diff
@@ -136,6 +136,63 @@ client.userPlaylists.default = {
     savedPlaylists: []
 };
 
+client.currency = new Enmap({
+    name: `currency`,
+    fetchAll: false,
+    autoFetch: true,
+    cloneLevel: `deep`
+});
+
+client.currency.default = {
+    balance: 0,
+    dailyStreak: 0,
+    dailyLastClaim: null,
+    dailyStreakScaling: 0.5
+};
+
+/**
+ * Adds/subtracts an integer amount from a user's balance
+ * @param {Integer} userID The ID of the user to affect 
+ * @param {Integer} amount The amount of money to add/subtract
+ * @returns A boolean representing success/failure
+ */
+client.currency.add = (userID, amount) => {
+    const userCurrency = client.currency.ensure(userID, client.currency.default);
+    if (isNaN(parseInt(amount))) {
+        return false;
+    } else {
+        if (amount < 0 && userCurrency[`balance`] + parseInt(amount) < 0) {
+            // Protect against negative balance
+            client.currency.set(userID, 0, `balance`);
+        } else if (userCurrency[`balance`] + parseInt(amount) > 999999999999) {
+            // Protect against ridiculously high amounts of money
+            client.currency.set(userID, 999999999999, `balance`);
```

Comment:
> __[Secret Scan]__ It may be dangerous to commit the AWS account ID. ([view](https://sider.review/gh/repos/219657356/pulls/109#issue-6525957))
>
> <details>
>
> AWS account ID is a credential (12-digit number). Once this leaks, an attacker may identify essential information
> on your AWS account, such as roles, users, etc.
>
> See also:
> * https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html
> * https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html
> * https://rhinosecuritylabs.com/aws/assume-worst-aws-assume-role-enumeration/
>
> | Rule | Severity |
> | ---- | -------- |
> | `sider.secrets.aws.account_id` | `warning` |
>
> You can [close this issue](https://sider.review/gh/repos/219657356/pulls/109#issue-6525957) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I closed this issue. Reason:
- When this is not an AWS account ID.